app/models/order_detail.rb

```diff
@@ -107,6 +108,28 @@ def journal_or_statement_date
   validate :account_usable_by_order_owner?, if: ->(o) { o.account_id_changed? || o.order.nil? || o.order.ordered_at.nil? }
   validates_length_of :note, maximum: 1000, allow_blank: true, allow_nil: true
   validate :valid_manual_fulfilled_at
+  validate :require_pricing_note
+
+  def require_pricing_note
+    return unless @manually_priced && SettingsHelper.feature_on?(:price_change_reason_required)
+    return if cost_estimated?
+
+    if price_change_reason.blank? && actual_costs_differ_from_calculated?
+      errors.add(:price_change_reason, :blank)
+    end
+  end
+
+  def actual_costs_differ_from_calculated?
+    dup_od = dup
+    dup_od.reservation = reservation.dup
+    dup_od.assign_price_policy
+
+    dup_od.actual_cost != actual_cost || dup_od.actual_subsidy != actual_subsidy
+  end
+
+  def actual_costs_match_calculated?
+    !actual_costs_differ_from_calculated?
```

Comment:
Would it be easier to read if we moved the logic into this method with `dup_od.actual_cost == actual_cost && dup_od.actual_subsidy == actual_subsidy` as the boolean logic and then made `actual_costs_differ_from_calculated?` the `!` of this? It's very minor, so it's not a big deal.